scripts/cluster_topology_observer.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Cluster topology changes observer."""
+
+import json
+import ssl
+import subprocess
+import sys
+from time import sleep
+from urllib.request import urlopen
+
+API_REQUEST_TIMEOUT = 5
+PATRONI_CLUSTER_STATUS_ENDPOINT = "cluster"
+
+# File path for the spawned cluster topology observer process to write logs.
+LOG_FILE_PATH = "/var/log/cluster_topology_observer.log"
+
+ssl._create_default_https_context = ssl._create_unverified_context
+
+
+def dispatch(run_cmd, unit, charm_dir):
+    """Use the input juju-run command to dispatch a :class:`ClusterTopologyChangeEvent`."""
+    dispatch_sub_cmd = "JUJU_DISPATCH_PATH=hooks/cluster_topology_change {}/dispatch"
+    # Input is generated by the charm
+    subprocess.run([run_cmd, "-u", unit, dispatch_sub_cmd.format(charm_dir)])  # noqa: S603
+
+
+def main():
+    """Main watch and dispatch loop.
+
+    Watch the Patroni API cluster info. When changes are detected, dispatch the change event.
+    """
+    patroni_url, run_cmd, unit, charm_dir = sys.argv[1:]
```

Comment:
Validate is removed, since cert validation is generally disabled. I don't think we need to maintain args compatibility.